### DISCUSSION TITLE
코딩 컨벤션과 빌더 vs 생성자

### DISCUSSION THREAD

**Discussion by Bue-von-hon** [f-lab-edu/foody-lab#24] [Ideas]
프로젝트 업데이트 받아서 확인해보니 코딩 컨벤션에 차이가 크네요:sob::sob:
Google 스타일 컨벤션에 탭 4, 인덴트 4, 연속 인덴트 4로 맞추면 서로 다른 부분을 교정할 수 있어 제안해봅니다..!

다만 Order객체의 of메소드에서 파라미터로 받는 값들이 너무 많더라구요
각각 의미가 달라서 파라미터로 받는 값들 중 일부를 클래스로 묶기도 곤란하고, 파라미터들의 타입도 제각각이라 빌더 패턴이 효과적일지 의문이더라구요..
(개인적으로 빌더 패턴은 파라미터들 타입이 많이 겹치면 좋더라구요)
메소드 파라미터가 4개 이상일때 줄바꿈하는게 잘 읽히기는 하는데 어떤지 모르겠네요..?

p.s. [여기서](https://sas-study.tistory.com/445) 컨벤션 적용하는 방법을 확인할 수 있어요!

**Comment by Taewoongjung** [comment]
세팅을 했는데 다른점이 있었나봐요.. 저 세팅으로 합시다!

개인적으로 빌더패턴이 괜찮다고 생각합니다. 파라미터 타입이 제각각일 때는 빌더 패턴이든 아니든 신경써야할 부분은 같다고 생각하고요.
빌더 패턴은 불변성 측면에서도 좋고, 파라메터가 많아졌을 때 가독성 측면에서도 좋다고 생각합니다. 
꼭 답이 있는 문제는 아닌 것 같고, 상황과 용도에 맞게 더 fit 한 생성자를 사용하는게 좋다고 생각합니다. 빌더든 아니든 방법론이지 각각의 장단점을 확실하게 인지하고 상황에 맞는 방법을 유연하게 적용할 수 있는 능력이 정말 중요한 것 같다고 생각합니다.
